koku/masu/external/downloader/ocp/ocp_report_downloader.py

```diff
@@ -66,35 +66,38 @@ def divide_csv_daily(file_path: str, manifest_id: int):
             manifest.report_tracker[file_prefix] = counter + 1
             manifest.save(update_fields=["report_tracker"])
         day_file = f"{file_prefix}.{counter}.csv"
-        day_filepath = f"{directory}/{day_file}"
+        day_filepath = file_path.parent.joinpath(day_file)
         df.to_csv(day_filepath, index=False, header=True)
         daily_files.append(
-            {"filename": day_file, "filepath": day_filepath, "date": datetime.datetime.strptime(day, "%Y-%m-%d")}
+            {
+                "filepath": day_filepath,
+                "date": datetime.strptime(day, "%Y-%m-%d"),
+                "num_hours": len(df.interval_start.unique()),
+            }
         )
     return daily_files
 
 
-def create_daily_archives(tracing_id, account, provider_uuid, filename, filepath, manifest_id, start_date, context={}):
+def create_daily_archives(tracing_id, account, provider_uuid, filepath, manifest_id, start_date, context={}):
     """
     Create daily CSVs from incoming report and archive to S3.
 
     Args:
         tracing_id (str): The tracing id
         account (str): The account number
         provider_uuid (str): The uuid of a provider
-        filename (str): The OCP file name
-        filepath (str): The full path name of the file
+        filepath (PosixPath): The full path name of the file
         manifest_id (int): The manifest identifier
         start_date (Datetime): The start datetime of incoming report
         context (Dict): Logging context dictionary
     """
     manifest = CostUsageReportManifest.objects.get(id=manifest_id)
-    daily_file_names = []
+    daily_file_names = {}
```

Comment:
This is a bigger picture idea I just wanted to mention. It may be beneficial here to use a class instead of a dictionary for storing structured data.

Dictionaries are great, but we hit their limits really fast when the data gets complicated. Storing data in classes is a lot more precise. Maybe there is value in ensuring the metadata fields are formatted consistently. Both `attrs` and `pydantic` have validation/coercion features built in that are helpful for ensuring uniform data.

Just food for thought.

Reply:
Right, this is absolutely a stop-gap until we get around to refactoring parts of our pipeline. I do not like this implementation because it's so unclear. But I'm working with what we have, and we need to get this out.